Traduzir Security >4.1 Redirection and Files (#815)
campuscode/rails-guides-pt-BR: pt-BR/security.md

[`config.action_controller.default_protect_from_forgery`]: configuring.html#config-action-controller-default-protect-from-forgery

Redirecionamento e Arquivos
---------------------

> Outra classe de vulnerabilidades de securanças envolve o uso de redirecionamento e arquivos em aplicações web.

Reviewer: ```suggestion
Outra classe de vulnerabilidades de seguranças envolve o uso de redirecionamento e arquivos em aplicações web.
```

Achei um typo, você poderia olhar por favor? :)

### Redirecionamento

WARNING: _Redirection in a web application is an underestimated cracker tool: Not only can the attacker forward the user to a trap website, they may also create a self-contained attack._

> Quando quer que o usuário é permitido passar (partes de) a URL para redirecionamento, é possívelmente vulnerável. O ataque mais óbvio seria redirecionar os usuários para uma aplicação web falsa que parece exatamente com a original. Isso também chamado de ataque de _phishing_ funciona por enviar um link não malicioso em um email para os usuários, injetando o link por _XSS_ na aplicação web ou colocando o link em um site externo. É não-suspeito, porque o link começa com a URL da aplicação web e a URL para o site malicioso é escondida no parâmetro de redirecionamento: http://www.example.com/site/redirect?to=www.attacker.com. Aqui está um exemplo com uma ação legada:

Reviewer 2: ```suggestion
Quando quer que o usuário é permitido passar (partes de) a URL para redirecionamento, é possivelmente vulnerável. O ataque mais óbvio seria redirecionar os usuários para uma aplicação web falsa que parece exatamente com a original. Isso também chamado de ataque de _phishing_ funciona por enviar um link não malicioso em um email para os usuários, injetando o link por _XSS_ na aplicação web ou colocando o link em um site externo. É não-suspeito, porque o link começa com a URL da aplicação web e a URL para o site malicioso é escondida no parâmetro de redirecionamento: http://www.example.com/site/redirect?to=www.attacker.com. Aqui está um exemplo com uma ação legada:
```

> Quando quer que o usuário é permitido passar (partes de) a URL para redirecionamento, é possivelmente vulnerável. O ataque mais óbvio seria redirecionar os usuários para uma aplicação web falsa que parece exatamente com a original. Isso também chamado de ataque de _phishing_ funciona por enviar um link não malicioso em um email para os usuários, injetando o link por _XSS_ na aplicação web ou colocando o link em um site externo. É não-suspeito, porque o link começa com a URL da aplicação web e a URL para o site malicioso é escondida no parâmetro de redirecionamento: http://www.example.com/site/redirect?to=www.attacker.com. Aqui está um exemplo com uma ação legada:

Reviewer 3: ```suggestion
Quando o usuário é permitido passar a (ou partes de uma) URL para redirecionamento, é passível de vulnerábilidades. O ataque mais óbvio seria redirecionar os usuários para uma aplicação web falsa que parece exatamente com a original. Isso também chamado de ataque de _phishing_ funciona por enviar um link não malicioso em um email para os usuários, injetando o link por _XSS_ na aplicação web ou colocando o link em um site externo. É não-suspeito, porque o link começa com a URL da aplicação web e a URL para o site malicioso é escondida no parâmetro de redirecionamento: http://www.example.com/site/redirect?to=www.attacker.com. Aqui está um exemplo com uma ação legada:
```

> Quando o usuário é permitido passar a (ou partes de uma) URL para redirecionamento, é passível de vulnerábilidades. O ataque mais óbvio seria redirecionar os usuários para uma aplicação web falsa que parece exatamente com a original. Isso também chamado de ataque de _phishing_ funciona por enviar um link não malicioso em um email para os usuários, injetando o link por _XSS_ na aplicação web ou colocando o link em um site externo. É não-suspeito, porque o link começa com a URL da aplicação web e a URL para o site malicioso é escondida no parâmetro de redirecionamento: http://www.example.com/site/redirect?to=www.attacker.com. Aqui está um exemplo com uma ação legada:

Reviewer: ```suggestion
Quando o usuário é permitido passar a (ou partes de uma) URL para redirecionamento, é passível de vulnerabilidades. O ataque mais óbvio seria redirecionar os usuários para uma aplicação web falsa que parece exatamente com a original. Isso também chamado de ataque de _phishing_ funciona por enviar um link não malicioso em um email para os usuários, injetando o link por _XSS_ na aplicação web ou colocando o link em um site externo. É não-suspeito, porque o link começa com a URL da aplicação web e a URL para o site malicioso é escondida no parâmetro de redirecionamento: http://www.example.com/site/redirect?to=www.attacker.com. Aqui está um exemplo com uma ação legada:
```

Reviewer: Outro ponto: acho que podemos traduzir `unsuspecting` nesse caso como `malicioso`, já que o vetor dos ataques de phishing são links que parecem inofensivos mas apontam para sites maliciosos. Me parece a maneira mais direta de traduzir. 

O que vc acha?

```ruby

> def legada

Reviewer 2: ```suggestion
def legacy
```

Normalmente não traduzimos código por ser mais comum na comunidade Rails os códigos serem em inglês. O que acha?

  redirect_to(params.update(action:'main'))
end
```

Isso redirecionará o usuário para a ação principal se ele tentou acessar uma ação legada. A intenção foi preservar os parâmetros da URL para a ação legada e passar eles para a ação principal. No entanto, pode ser explorado pelo atacante se eles incluíram uma chave _host_ na URL:

```
http://www.example.com/site/legacy?param1=xy&param2=23&host=www.attacker.com
```

> Se é no fim da URL, isso dificilmente será notado e redireciona o usuário para o _host_ `attacker.com`. Como uma regra geral, passar entradas de usuário diretamente no `redirect_to` é considerado perigoso. Uma contramedida simple seria _incluir apenas os parâmetros esperados em uma ação legada. (novamente uma abordagem de lista permitida, ao invés de remover parâmetros inesperados. E se você redirecionar para uma URL, cheque com uma lista permitida ou uma expressão regular.

Reviewer 3: Acho que tem um parenteses faltando aqui.
```suggestion
Se é no fim da URL, isso dificilmente será notado e redireciona o usuário para o _host_ `attacker.com`. Como uma regra geral, passar entradas de usuário diretamente no `redirect_to` é considerado perigoso. Uma contramedida simple seria _incluir apenas os parâmetros esperados em uma ação legada. (novamente uma abordagem de lista permitida, ao invés de remover parâmetros inesperados). E se você redirecionar para uma URL, cheque com uma lista permitida ou uma expressão regular.
```
